fbfmaproom/fbfmaproom.py

```diff
@@ -72,33 +72,42 @@
 
 APP.layout = fbflayout.app_layout()
 
-def table_columns(obs_config, obs_dataset_key):
-    def bad_year_css(row):
-        r = row['bad_year']
-        if pd.notna(r):
-            if r == "Bad":
-                return "cell-bad-year"
-            else:
-                return "cell-good-year"
-        else:
-            return ""
+
+def table_columns(obs_config, obs_dataset_keys):
 
     obs_dataset_names = {k: v["label"] for k, v in obs_config.items()}
     tcs = OrderedDict()
-    tcs["year_label"] = dict(name="Year", dynamic=None, style=None,
+    tcs["year_label"] = dict(name="Year", style=None,
                              tooltip="The year whose forecast is displayed on the map")
-    tcs["enso_state"] = dict(name="ENSO State", dynamic=None,
+    tcs["enso_state"] = dict(name="ENSO State",
                              tooltip="Displays whether an El Niño, Neutral, or La Niña state occurred during the year",
                              style=lambda row: {'El Niño': 'cell-el-nino',
                                                 'La Niña': 'cell-la-nina',
                                                 'Neutral': 'cell-neutral'}.get(row['enso_state'], ""))
-    tcs["forecast"] = dict(name="Forecast, %", dynamic=None,
+    tcs["forecast"] = dict(name="Forecast, %",
                            tooltip="Displays all the historical flexible forecast for the selected issue month and location",
                            style=lambda row: "cell-severity-" + str(row['severity']) if row['worst_pnep'] == 1 else "")
-    tcs["obs_rank"] = dict(name=f"{obs_dataset_names[obs_dataset_key]} Rank",
-                           style=lambda row: "cell-severity-" + str(row['severity']) if row['worst_obs'] == 1 else "",
-                           tooltip=None, dynamic=None)
-    tcs["bad_year"] = dict(name="Reported Bad Years", dynamic=None,
+
+    def make_obs_column(obs_key):
+        return dict(
+            name=f"{obs_dataset_names[obs_key]} Rank",
+            style=lambda row: "cell-severity-" + str(row['severity']) if row[f'worst_{obs_key}'] == 1 else "",
```

Comment:
Side note: I am pleased with how the fbftable component has worked given that, since we're using these lambdas to control the style logic, we didn't actually have to modify the component itself. I guess that's a sign the abstraction actually worked.